core/src/main/scala/kafka/tools/JmxTool.scala

```diff
@@ -122,7 +125,26 @@ object JmxTool extends Logging {
       else
         List(null)
 
-    val names = queries.flatMap((name: ObjectName) => mbsc.queryNames(name, null).asScala)
+    val hasPatternQueries = queries.exists((name: ObjectName) => name.isPattern)
+
+    retries = 0
+    var names : Iterable[ObjectName] = null
+    if (!hasPatternQueries) {
+      do {
+        if (names != null) {
+          System.err.println("Could not find all object names, retrying")
+          retries += 1
+          Thread.sleep(500)
```

Comment:
It could be hard-coded. I was just thinking it is a more intuitive way to achieve the same result. A bit weird to have to increase the backoff to make sure we're giving enough time to the tool.